apps/hash-frontend/src/pages/[shortname].page/edit-pinned-entity-types-modal.tsx

```diff
@@ -0,0 +1,424 @@
+import {
+  AsteriskRegularIcon,
+  IconButton,
+  Modal,
+} from "@hashintel/design-system";
+import { types } from "@local/hash-isomorphic-utils/ontology-types";
+import { EntityTypeWithMetadata, OwnedById } from "@local/hash-subgraph";
+import { extractBaseUrl } from "@local/hash-subgraph/type-system-patch";
+import {
+  Box,
+  Divider,
+  ModalProps,
+  Typography,
+  typographyClasses,
+} from "@mui/material";
+import {
+  FunctionComponent,
+  ReactElement,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  DragDropContext,
+  Draggable,
+  DraggableProvided,
+  DraggingStyle,
+  Droppable,
+  NotDraggingStyle,
+  OnDragEndResponder,
+} from "react-beautiful-dnd";
+import { createPortal } from "react-dom";
+import { useFieldArray, useForm } from "react-hook-form";
+
+import { useBlockProtocolUpdateEntity } from "../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-update-entity";
+import { useBlockProtocolCreateEntityType } from "../../components/hooks/block-protocol-functions/ontology/use-block-protocol-create-entity-type";
+import { Org, User } from "../../lib/user-and-org";
+import { useEntityTypesContextRequired } from "../../shared/entity-types-context/hooks/use-entity-types-context-required";
+import { ArrowUpRightRegularIcon } from "../../shared/icons/arrow-up-right-regular-icon";
+import { CustomLinkIcon } from "../../shared/icons/custom-link-icon";
+import { GripDotsVerticalRegularIcon } from "../../shared/icons/grip-dots-vertical-regular-icon";
+import { PlusRegularIcon } from "../../shared/icons/plus-regular";
+import { XMarkRegularIcon } from "../../shared/icons/x-mark-regular-icon";
+import { Button, Link } from "../../shared/ui";
+import { entityTypeIcons } from "../../shared/use-entity-icon";
+import { EntityTypeSelector } from "../[shortname]/entities/[entity-uuid].page/create-entity-page/entity-type-selector";
```

Comment:
This should be in a shared folder

Reply:
Addressed in https://github.com/hashintel/hash/pull/3338/commits/579e8cfa9b036947cb67ab568dbb6435b96ee96f